lib/stylesheet/manager.rb

```diff
@@ -70,20 +70,28 @@ def self.stylesheet_details(target = :desktop, media = 'all', theme_ids = :missi
 
     @lock.synchronize do
       stylesheets = []
+
+      themes = Theme.where(id: theme_ids).each_with_object({}) do |theme, hash|
+        hash[theme.id] = theme
+      end
+
+      theme_scss_checker = Theme::ScssChecker.new(target, theme_ids)
+
       theme_ids.each do |theme_id|
         data = { target: target }
         cache_key = "path_#{target}_#{theme_id}_#{current_hostname}"
         href = cache[cache_key]
 
         unless href
           builder = self.new(target, theme_id)
+          builder.theme = themes[theme_id]
```

Comment:
Instead of waiting for `Theme.find_by` to be called on the `theme_id`, we eager load stuff and just assign it here.